x-pack/plugins/apm/server/lib/search_strategies/correlations/get_query_with_params.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { estypes } from '@elastic/elasticsearch';
+import {
+  TRANSACTION_DURATION,
+  TRANSACTION_NAME,
+  PROCESSOR_EVENT,
+  SERVICE_NAME,
+} from '../../../../common/elasticsearch_fieldnames';
+import type { SearchServiceParams } from '../../../../common/search_strategies/correlations/types';
+import {
+  environmentQuery as getEnvironmentQuery,
+  rangeQuery as getRangeQuery,
+} from '../../../utils/queries';
+
+export enum ProcessorEvent {
+  transaction = 'transaction',
+  error = 'error',
+  metric = 'metric',
+  span = 'span',
+  profile = 'profile',
+}
```

Comment:
This look like a duplicate of:
https://github.com/elastic/kibana/blob/9a1f5a4a7a81603f1b11ff722c3c1c3707e00ca9/x-pack/plugins/apm/common/processor_event.ts#L8

Reply:
Updated here https://github.com/elastic/kibana/pull/99905/commits/7dee7b41cc2dd243dcde7e5a5660c9e5b40532c3